### PR TITLE
expression: implement vectorized evaluation for `builtinOctStringSig`

### PR DESCRIPTION
**Pull request by mmyj**
### What problem does this PR solve? 
implement vectorized evaluation for builtinOctStringSig , for #12106

### What is changed and how it works?
```
go test -v -benchmem -bench=BenchmarkVectorizedBuiltinStringFunc -run=BenchmarkVectorizedBuiltinStringFunc -args 'builtinOctStringSig'
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinStringFunc/builtinOctStringSig-VecBuiltinFunc-4                  32122             37473 ns/op             466 B/op        233 allocs/op
BenchmarkVectorizedBuiltinStringFunc/builtinOctStringSig-NonVecBuiltinFunc-4               19594             60872 ns/op             466 B/op        233 allocs/op
PASS
ok      github.com/pingcap/tidb/expression      3.430s
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_string_vec.go [hunk 1]
```diff
@@ -680,11 +680,34 @@ func (b *builtinFieldRealSig) vecEvalInt(input *chunk.Chunk, result *chunk.Colum
...
+		if buf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+		str := buf.GetString(i)
+		val, err := strconv.ParseUint(str, 10, 64)
+		if err != nil {
+			return err
+		}
+		result.AppendString(strconv.FormatUint(val, 8))
```
Comment (XuHuaiyu) [inline expression/builtin_string_vec.go]: This implementation is not totally the same as the non-vec version?

File: expression/builtin_string_vec.go [hunk 2]
```diff
@@ -680,11 +680,34 @@ func (b *builtinFieldRealSig) vecEvalInt(input *chunk.Chunk, result *chunk.Colum
...
+		return err
+	}
+
+	result.ReserveString(n)
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+		str := buf.GetString(i)
```
Comment (qw4990) [inline expression/builtin_string_vec.go]: Should we check the negative sign here?

Comment (mmyj) [reply expression/builtin_string_vec.go]: I will fix it.

**Comment by mmyj** [comment]
@qw4990  I have checked the negative sign.

File: expression/builtin_string_vec.go [hunk 3]
```diff
@@ -789,11 +789,49 @@ func (b *builtinFieldRealSig) vecEvalInt(input *chunk.Chunk, result *chunk.Colum
...
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+		negative, overflow := false, false
+		str := buf.GetString(i)
+		str = getValidPrefix(strings.TrimSpace(str), 10)
+		if len(str) == 0 {
+			result.AppendString("0")
```
Comment (qw4990) [inline expression/builtin_string_vec.go]: Should we add a `continue` statement below this line?

Comment (mmyj) [reply expression/builtin_string_vec.go]: I will fix it.

**Comment by mmyj** [comment]
@qw4990 PTAL

**Pull request merged**